core/balances.go

```diff
@@ -264,6 +264,11 @@ func (bc *BaseContract) TokenBalanceGetLocked(address *types.Address) (*big.Int,
 	return new(big.Int).SetBytes(balance.Bytes()), err
 }
 
+func (bc *BaseContract) AllowedBalanceGetLocked(token string, address *types.Address) (*big.Int, error) {
+	balanceValue, err := balance.Get(bc.stub, balance.BalanceTypeAllowedLocked, address.String(), token)
+	return new(big.Int).SetBytes(balanceValue.Bytes()), err
+}
+
```

Comment:
Please move the function between AllowedIndustrialBalanceSub and AllowedBalanceLock